braindecode/models/eegconformer.py

```diff
@@ -0,0 +1,313 @@
+# Authors: Yonghao Song <eeyhsong@gmail.com>
+#
+# License: BSD (3-clause)
+
+import torch
+import torch.nn.functional as F
+from einops import rearrange
+from einops.layers.torch import Rearrange, Reduce
+from torch import nn, Tensor
+
+
+class EEGConformer(nn.Module):
+    """EEG Conformer.
+
+    Convolutional Transformer for EEG decoding.
+
+    The paper and original code with more details about the methodological
+    choices are available at the [Song2022]_ and [ConformerCode]_.
+
+    This neural network architecture recieves a traditional braindecode input.
+    The input shape should be three-dimensional matrix representing the EEG
+    signals.
+
+         (batch_size, n_channels, n_timesteps)`.
+
+    The EEG Conformer architecture is composed of three modules:
+        - PatchEmbedding
+        - TransformerEncoder
+        - ClassificationHead
+
+    Notes
+    -----
+    The authors recommend using augment data before using Conformer, e.g. S&R,
+    at the end of the code.
+    Please refer to the original paper and code for more details.
+
+    .. versionadded:: 0.8
+
+    We aggregate the parameters based on the parts of the models, or
+    when the parameters were used first, e.g. n_filters_conv.
+
+    Parameters
+    ----------
+    - n_filters_conv: int
+        Length of kernels for the temporal convolution layer (first layer).
+    - n_filters_time: int
+        Number of temporal filters.
+    - filter_time_length: int
+        Length of the temporal filter.
+    - n_channels: int
+        Number of channels to be used as number of spatial filters.
+    - pool_time_length: int
+        Length of temporal poling filter.
+    - pool_time_stride: int
+        Length of stride between temporal pooling filters.
+    - drop_prob: float
+        Dropout rate of the convolutional layer.
+    - att_depth: int
+        Number of self-attention layers.
+    - att_heads: int
+        Number of attention heads.
+    - att_drop_prob: float
+        Dropout rate of the self-attention layer.
+    - final_fc_length: int
+        The dimension of the fully connected layer.
+    - n_classes: int
+        Number of classes to predict (number of output filters of last layer).
+
+    References
+    ----------
+    .. [Song2022] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       IEEE Transactions on Neural Systems and Rehabilitation Engineering,
+       31, pp.710-719. https://ieeexplore.ieee.org/document/9991178
+    .. [ConformerCode] Song, Y., Zheng, Q., Liu, B. and Gao, X., 2022. EEG
+       conformer: Convolutional transformer for EEG decoding and visualization.
+       https://github.com/eeyhsong/EEG-Conformer.
+    """
+
+    def __init__(
+            self,
+            n_classes,
+            n_channels,
+            n_filters_conv=40,
+            n_filters_time=25,
+            filter_time_length=25,
+            n_kernel_avg_pool=75,
```

Comment:
`n_kernel_avg_pool` is missing in the docstring - I would suggest sorting the params in the docstring according to the order in the code